lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1043,31 +1039,24 @@ export class UserAgentApplication {
         } else {
             // Redirect cases
             tokenResponseCallback = null;
+            window.location.hash = "";
             // if set to navigate to loginRequest page post login
             if (this.config.auth.navigateToLoginRequestUrl) {
                 this.cacheStorage.setItem(TemporaryCacheKeys.URL_HASH, locationHash);
                 if (window.parent === window) {
                     const loginRequestUrl = this.cacheStorage.getItem(`${TemporaryCacheKeys.LOGIN_REQUEST}${Constants.resourceDelimiter}${stateInfo.state}`, this.inCookie);
+                    const currentUrl = this.getCurrentUrl();
 
                     // Redirect to home page if login request url is null (real null or the string null)
                     if (!loginRequestUrl || loginRequestUrl === "null") {
                         this.logger.error("Unable to get valid login request url from cache, redirecting to home page");
                         window.location.href = "/";
-                    } else {
-                        window.location.href = loginRequestUrl;
+                    } else if (currentUrl != loginRequestUrl) {
```

Comment:
```suggestion
                    } else if (currentUrl !== loginRequestUrl) {
```

Reply:
Triple (not) equals is preferred over double equals, as it doesn't perform type coercion.